app/uk/gov/hmrc/breathingspaceifproxy/connector/ConnectorHelper.scala

```diff
@@ -18,44 +18,87 @@ package uk.gov.hmrc.breathingspaceifproxy.connector
 
 import scala.concurrent.Future
 
+import cats.data._
+import cats.syntax.option._
 import play.api.Logging
-import play.api.http.{Status => HttpStatus}
+import play.api.http.{HeaderNames, Status => HttpStatus}
+import play.api.http.Status.{CREATED, NOT_FOUND, OK}
 import play.api.mvc.Result
 import play.api.mvc.Results.Status
 import play.mvc.Http.MimeTypes
 import uk.gov.hmrc.breathingspaceifproxy._
 import uk.gov.hmrc.breathingspaceifproxy.model._
+import uk.gov.hmrc.breathingspaceifproxy.model.BaseError.RESOURCE_NOT_FOUND
 import uk.gov.hmrc.http._
 
+// TODO: should not be composing responses to PEGA here in the IF connector!
 trait ConnectorHelper extends HttpErrorFunctions with Logging {
 
-  def composeResponseFromIF(response: HttpResponse)(implicit url: Url, hc: HeaderCarrier): Result = {
+  val ifCorrelationIdHeaderName = "CorrelationId"
+  val ifRequestTypeHeaderName = "RequestType"
+  val ifStaffIdHeaderName = "StaffId"
+
+  def composeResponseFromIF(response: HttpResponse)(implicit url: Url, headerSet: RequiredHeaderSet): Future[Result] = {
     logResponse(response)
-    Status(response.status)(response.body)
-      .withHeaders(response.headers.toList.map(header => (header._1, header._2.mkString(""))): _*)
-      .as(MimeTypes.JSON)
+
+    def composeHappyResponse(response: HttpResponse) = Future.successful {
+      Status(response.status)(response.body)
+        .withHeaders(
+          (Header.CorrelationId, headerSet.correlationId.value),
+          (HeaderNames.CONTENT_TYPE, MimeTypes.JSON)
```

Comment:
I think this can be removed, no? You already have `.as(MimeTypes.JSON)`

Reply:
Not sure, that sets `body.as(contentType)` not the header. Could remove it but unsure what the difference is between the two content type flags